apps/fxc-front/src/app/logic/score/improvedScorer.ts

```diff
@@ -0,0 +1,231 @@
+
+const czlScoringRule = [
+  { ...openDistanceBase, multiplier: 1 },
+  { ...freeTriangleBase, multiplier: 1.8 },
+  { ...faiTriangleBase, multiplier: 2.2 },
+];
+
+const czeScoringRule = [
+  { ...openDistanceBase, multiplier: 1 },
+  { ...freeTriangleBase, multiplier: 1.2 },
+  { ...faiTriangleBase, multiplier: 1.4 },
+];
+
+const czoScoringRule = [
+  { ...openDistanceBase, multiplier: 0.8 },
+  { ...freeTriangleBase, multiplier: 1.2 },
+  { ...faiTriangleBase, multiplier: 1.4 },
+];
+
+const leoScoringRule = [
+  { ...openDistanceBase, multiplier: 1.5 },
+  { ...freeTriangleBase, multiplier: 1.75, closingDistanceRelative: 0.2 },
+  { ...faiTriangleBase, multiplier: 2, closingDistanceRelative: 0.2 },
+];
+
+const norScoringRule = [
+  { ...openDistanceBase, multiplier: 1 },
+  { ...freeTriangleBase, multiplier: 1.7, closingDistanceRelative: 0.05 },
+  { ...freeTriangleBase, multiplier: 1.5, closingDistanceRelative: 0.2 },
+  { ...faiTriangleBase, multiplier: 2.4, closingDistanceRelative: 0.05 },
+  { ...faiTriangleBase, multiplier: 2.2, closingDistanceRelative: 0.2 },
+];
+
+const ukcScoringRule = [
+  { ...openDistanceBase, multiplier: 1, minDistance: 5 },
+  { ...outAndReturnBase, multiplier: 1.2, minDistance: 5 },
+  { ...outAndReturnBase, multiplier: 1.3, minDistance: 15 },
+  { ...outAndReturnBase, multiplier: 1.7, minDistance: 35 },
+  { ...freeTriangleBase, multiplier: 1.2, minDistance: 5 },
+  { ...freeTriangleBase, multiplier: 1.3, minDistance: 15 },
+  { ...freeTriangleBase, multiplier: 1.7, minDistance: 35 },
+  { ...faiTriangleBase, multiplier: 1.5, minDistance: 5 },
+  { ...faiTriangleBase, multiplier: 1.7, minDistance: 15 },
+  { ...faiTriangleBase, multiplier: 2, minDistance: 35 },
+];
+
+const ukiScoringRule = [
+  { ...openDistanceBase, multiplier: 1, minDistance: 10 },
+  { ...outAndReturnBase, multiplier: 1.2, minDistance: 35 },
+  { ...freeTriangleBase, multiplier: 1.2, minDistance: 35 },
+  { ...faiTriangleBase, multiplier: 1.5, minDistance: 25 },
+];
+
+const uknScoringRule = [
+  { ...openDistanceBase, multiplier: 1, minDistance: 10 },
+  { ...outAndReturnBase, multiplier: 1.3, minDistance: 15 },
+  { ...outAndReturnBase, multiplier: 1.7, minDistance: 35 },
+  { ...freeTriangleBase, multiplier: 1.3, minDistance: 15 },
+  { ...freeTriangleBase, multiplier: 1.7, minDistance: 35 },
+  { ...faiTriangleBase, multiplier: 1.7, minDistance: 15 },
+  { ...faiTriangleBase, multiplier: 2, minDistance: 25 },
+];
+
+const xcppgScoringRule = [
+  { ...openDistanceBase, multiplier: 1 },
+  { ...freeTriangleBase, multiplier: 2, closingDistanceFixed: 0.8 },
+  { ...faiTriangleBase, multiplier: 4, closingDistanceFixed: 0.8 },
+];
+
+const wxcScoringRule = [
+  { ...openDistanceBase, multiplier: 1 },
+  { ...freeTriangleBase, multiplier: 1.75, closingDistanceRelative: 0.2 },
+  { ...faiTriangleBase, multiplier: 2, closingDistanceFixed: 0.2 },
+];
+
+const leaguesScoringRules: Map<LeagueCode, object> = new Map()
```

Comment:
You can use
```
new Map([
 [k1, v1],
 [k2, v2],
 ...
]);
```

Reply:
thanks. I have many short comings with typescript and JS.